### PR TITLE
fix: Fix/10812 redundant source map upload

### LINKED ISSUE #10812: Prevent redundant Sentry sourcemap uploads
### What is this about?

This task tracks preventing redundant Sentry sourcemap uploads. For context: We've noticed that there are duplicate sourcemaps being uploaded for releases under the same distribution name. This messes with stack traces since production bugs can now be associated with sourcemaps generated with different stack traces. In turn, this can make debugging Sentry issues impossible. The solution is to fix the system so that we only upload one set of source maps to each platform for each distributed version. 

This shows redundant sourcemaps uploaded to the same distribution name
image

### Scenario

_No response_

### Design

_No response_

### Technical Details

- It is believed that the issue occurs because certain release flows are used indiscriminately
- For example, `build_ios_release`/`build_android_release` is a step used for cutting RCs as well as uploading builds to TestFlight for testing 
  - This pipeline should be renamed/created to describe the building and staging of apps in test disruptions
  - `release_builds_to_store_pipeline` - remove this pipeline
  -  `build_send_apps_to_testflight` - create a pipeline that disables source map uploads
- In the scenario above, it's possible that `build_ios_release`/`build_android_release` overwrites the sourcemaps for the RC, which eventually becomes prod
- The technical task here is to confirm this suspicion and possibly abstract + designate a flow only used for cutting RCs. Once this flow is created, we need to ensure that only this flow uploads Sentry sourcemaps

### Threat Modeling Framework

_No response_

### Acceptance Criteria

- Only a single set of sourcemaps is being uploaded to each RC/distribution version
- Reported errors in Sentry shows the correct stack trace
- The new abstracted flow creates an RC + uploads the correct sourcemaps in Sentry

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by Cal-L**
## **Description**

This PR introduces two new workflows in Bitrise
- `build_android_release_and_upload_sourcemaps` - Runs `build_android_release` and uploads sourcemaps to Sentry
- `build_ios_release_and_upload_sourcemaps` - Runs `build_ios_release` and uploads sourcemaps to Sentry

By default, both `build_android_release` and `build_ios_release` will not automatically upload sourcemaps to Sentry

Related PR - https://github.com/MetaMask/metamask-mobile/pull/11490

## **Related issues**

Fixes: #10812 

## **Manual testing steps**

To build release AND upload sourcemaps, use the two workflows in Bitrise:
- `build_android_release_and_upload_sourcemaps`
- `build_ios_release_and_upload_sourcemaps`

To only build release without uploading sourcemaps, use the two workflows in Bitrise:
- `build_android_release`
- `build_ios_release`

### **After**

In this test, we ran the default `build_ios_release` and `build_android_release`, which resulted in no sourcemaps being uploaded
- Android - https://app.bitrise.io/build/bc568645-d1e4-418c-bea0-6aafe8f2e21c
- iOS - https://app.bitrise.io/build/e7c224b3-c0f3-4be9-afa0-b3ad423d0da7

In this test, we enabled Sentry sourcemap upload and re-used `build_ios_release` and `build_android_release` to mimic the workflows that uploads sourcemaps. We can see in the two builds that it successfully uploads sourcemaps.
- Android - https://app.bitrise.io/build/d93989e0-3ca8-4bab-aaf2-48b0ca104986
- iOS - https://app.bitrise.io/build/b24edfed-b020-4ac1-bf44-191d284b099b

Sourcemaps uploaded
- Android:
Screenshot 2024-10-23 at 6 08 21 PM
- iOS:
Screenshot 2024-10-23 at 6 08 31 PM

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: scripts/build.sh [hunk 1]
```diff
@@ -300,7 +300,7 @@ buildIosRelease(){
   	remapEnvVariableRelease
 
 	# Enable Sentry to auto upload source maps and debug symbols
-	export SENTRY_DISABLE_AUTO_UPLOAD="false"
```
Comment (Cal-L) [inline scripts/build.sh:303]: This disables sentry upload by default

File: bitrise.yml [hunk 2]
```diff
@@ -1303,9 +1308,14 @@ workflows:
             - pipeline_intermediate_files: sourcemaps/ios/index.js.map:BITRISE_APP_STORE_SOURCEMAP_PATH
             - deploy_path: sourcemaps/ios/index.js.map
           title: Deploy Source Map
+  build_ios_release_and_upload_sourcemaps:
```
Comment (Cal-L) [inline bitrise.yml:1311]: New workflow, which also uploads sourcemaps. to be used by something like Runway

File: bitrise.yml [hunk 3]
```diff
@@ -1094,6 +1094,11 @@ workflows:
       bitrise.io:
         stack: linux-docker-android-22.04
         machine_type_id: elite-xl
+  build_android_release_and_upload_sourcemaps:
```
Comment (Cal-L) [inline bitrise.yml:1097]: New workflow, which also uploads sourcemaps. to be used by something like Runway

**Review by NicolasMassart — APPROVED** [review]
Looks good to me. Thanks Cal for this!

**Pull request merged**